api/server/router/plugin/plugin.go

```diff
@@ -9,7 +14,8 @@ type pluginRouter struct {
 }
 
 // NewRouter initializes a new plugin router
-func NewRouter(b Backend) router.Router {
+func NewRouter() router.Router {
+	b := plugincontroller.Wait(context.Background())
```

Comment:
It would be a lot nicer if this could continue to receive it's dependency instead of having to wait for it to be available. I don't think the API routers should be aware of the components. 

We should try to avoid the component framework/registry leaking into every package. I think only a few central packages should be aware of components, and everything else should just receive references to their dependencies.